lib/services/database_mutation_functions.dart

```diff
@@ -205,6 +226,9 @@ class DataBaseMutationFunctions {
       for (var i in userInfo.beacon) {
         if (!beaconIds.contains(i.id)) {
           beaconIds.add(i.id);
+          //save in local db for future ref/use.
+          await hiveDb.putBeaconInBeaconBox(i.id, i);
+          i = hiveDb.beaconsBox.get(i.id);
```

Comment:
Because of this, fetching the beacon's list takes longer than expected.
Beacon's list stores the joined or created beacons, and we are already storing it when either it is being created or joined, then why do we need this?

Reply:
To fetch the latest data about the beacon and store it in local db.  
In this function( hiveDb.putBeaconInBeaconBox(i.id, i);), we fetch all beacon data, unlike in the homescreen query, where not all beacon data is fetched, for eg(landmarks are not fetched).
Maybe we can remove the await to make this happen in background.